folly/portability/Time.h

```diff
@@ -21,15 +21,38 @@
 
 #include <folly/portability/Config.h>
 
+#ifdef __APPLE__
+#include <Availability.h>
+#define FOLLY_HAVE_CLOCK_GETTIME_DEFINITION \
+    ((defined(__MAC_OS_X_VERSION_MAX_ALLOWED) && \
+      defined(__MAC_10_12) && \
+      __MAC_OS_X_VERSION_MAX_ALLOWED >= __MAC_10_12) || \
+     (defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && \
+      defined(__IPHONE_10_0) && \
+      __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_10_0) || \
+     (defined(__TV_OS_VERSION_MAX_ALLOWED) && \
+      defined(__TVOS_10_0) && \
+      __TV_OS_VERSION_MAX_ALLOWED >= __TVOS_10_0) || \
+     (defined(__WATCH_OS_VERSION_MAX_ALLOWED) && \
+      defined(__WATCHOS_3_0) && \
+      __WATCH_OS_VERSION_MAX_ALLOWED >= __WATCHOS_3_0))
+#else
+#define FOLLY_HAVE_CLOCK_GETTIME_DEFINITION FOLLY_HAVE_CLOCK_GETTIME
+#endif
+
 // These aren't generic implementations, so we can only declare them on
 // platforms we support.
 #if !FOLLY_HAVE_CLOCK_GETTIME && (defined(__MACH__) || defined(_WIN32))
+
+#if !FOLLY_HAVE_CLOCK_GETTIME_DEFINITION
 #define CLOCK_REALTIME 0
 #define CLOCK_MONOTONIC 1
 #define CLOCK_PROCESS_CPUTIME_ID 2
 #define CLOCK_THREAD_CPUTIME_ID 3
 
 typedef uint8_t clockid_t;
+#endif
+
 extern "C" int clock_gettime(clockid_t clk_id, struct timespec* ts);
```

Comment:
It would be better to check for the existence of `clock_gettime` in the configure script rather than inline.